### PR TITLE
Spells/DK: Fix proc of Frost Fever

### PR DESCRIPTION
**Pull request by aquadeus**
**Changes proposed:**

-  Fix proc of Frost Fever

**Tests performed:**

Tested in-game

### CODE REVIEW

File: src/server/scripts/Spells/spell_dk.cpp [hunk 1]
```diff
@@ -660,6 +660,25 @@ class spell_dk_festering_strike : public SpellScript
...
+class spell_dk_frost_fever_proc : public AuraScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellEffect({ { spellInfo->Id, EFFECT_0 } });
+    }
+
+    bool CheckProc(AuraEffect const* /*aurEff*/, ProcEventInfo const& /*eventInfo*/) const
+    {
+        return roll_chance_i(GetEffect(EFFECT_0)->GetAmount());
```
Comment (Nyr97) [inline src/server/scripts/Spells/spell_dk.cpp:673]: It's always recommendable to pass CalcAmount in case any effect modifies this amount

Comment (mdX7) [reply src/server/scripts/Spells/spell_dk.cpp:673]: that is for effects which aren't within same aura, theres no need to recalculate for no reason

Comment (Nyr97) [reply src/server/scripts/Spells/spell_dk.cpp:673]: but what if a different aura modifies base value of this aura?

Comment (mdX7) [reply src/server/scripts/Spells/spell_dk.cpp:673]: thats already taken into account in `GetAmount`

**Review by Shauren — APPROVED** [review]

**Pull request merged**